### PR TITLE
Map service integration

### PR DESCRIPTION
**Pull request by stawberryslayer**
Fix issues:
Map service conflict
Database initialization
Database suspend annotation

### CODE REVIEW

File: app/src/main/java/com/cs407/map_application/MainActivity.kt [hunk 1]
```diff
@@ -4,55 +4,67 @@ import android.content.Intent
...
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
+import androidx.lifecycle.lifecycleScope
+
 import com.google.android.gms.maps.GoogleMap
-import com.google.android.gms.maps.OnMapReadyCallback
-import com.google.android.gms.maps.model.LatLng
 
-class MainActivity : AppCompatActivity(), OnMapReadyCallback {
+class MainActivity : AppCompatActivity() {
```
Comment (stawberryslayer) [inline app/src/main/java/com/cs407/map_application/MainActivity.kt:16]: no map in main activity, so OnMapReadyCallback not needed here

File: app/src/main/java/com/cs407/map_application/MainActivity.kt [hunk 2]
```diff
@@ -4,55 +4,67 @@ import android.content.Intent
...
 
-    override fun onMapReady(map: GoogleMap) {
-        TODO("Not yet implemented")
-        googleMap = map
-        mapService = MapService(googleMap)
+        }
 
-        // Now you can use mapService to interact with the map
-        val defaultLocation = LatLng(37.7749, -122.4194) // Example coordinates
-        mapService.initializeMap(defaultLocation, 10f)
```
Comment (stawberryslayer) [inline app/src/main/java/com/cs407/map_application/MainActivity.kt:52]: Duplicate implementation, already have this in Details page

**Comment by LIU-Ziqi-77** [comment]
Looks good to me, you did a nice job!

**Review by LIU-Ziqi-77 — APPROVED** [review]
Nice job.

**Pull request merged**